src/main/java/org/spongepowered/api/GameRegistry.java

```diff
@@ -85,6 +85,23 @@
 public interface GameRegistry {
 
     /**
+     * Creates a catalog key.
+     *
+     * @param namespace The namespace
+     * @param value The value
+     * @return A new catalog key
+     */
+    CatalogKey createKey(final String namespace, final String value);
+
+    /**
+     * Resolves a catalog key.
+     *
+     * @param value The value
+     * @return A new catalog key
+     */
+    CatalogKey resolveKey(final String value);
```

Comment:
Perhaps this should accept a particular `CatalogType` class in the case where we have different defaults depending on the type? Like `BlockType` versus `DataRegistration` etc.

Reply:
I still believe we should be providing a type class to better resolve the key for, since you can have ambiguity in some cases, like the format of the block state id's compared to the block type id's.